tests/unit/amo/components/TestUserProfileEdit.js

```diff
@@ -0,0 +1,601 @@
+    const root = renderUserProfileEdit();
+
+    expect(root.find('.UserProfileEdit')).toHaveLength(1);
+    expect(root.find(Notice)).toHaveLength(0);
+
+    expect(root.find('.UserProfileEdit--Card').first())
+      .toHaveProp('header', 'Account');
+    expect(root.find('.UserProfileEdit-email--help')).toHaveLength(1);
+    expect(root.find('.UserProfileEdit-aside')).toHaveText(oneLine`Tell users a
+      bit more information about yourself. These fields are optional, but
+      they'll help other users get to know you better.`
+    );
+    expect(root.find({ htmlFor: 'biography' }))
+      .toHaveText('Introduce yourself to the community if you like');
+  });
+
+  it('dispatches fetchUserAccount action if username is not found', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const username = 'i-am-not-tofumatt';
+    const root = renderUserProfileEdit({ params: { username }, store });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username,
+    }));
+  });
+
+  it('does not dispatch fetchUserAccount action if there is no username', () => {
+    const { store } = signInUserWithUsername('tofumatt');
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    // This happens when loading the user edit profile page of the current
+    // logged user (e.g., page refresh).
+    renderUserProfileEdit({ params: {}, store });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('dispatches fetchUserAccount action if username changes', () => {
+    const username = 'black-panther';
+    const params = { username };
+
+    const { store } = signInUserWithUsername(username);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    root.setProps({ username: 'killmonger' });
+
+    sinon.assert.calledWith(dispatchSpy, fetchUserAccount({
+      errorHandlerId: root.instance().props.errorHandler.id,
+      username: 'killmonger',
+    }));
+
+    expect(root.find('.UserProfileEdit-location')).toHaveProp('value', '');
+  });
+
+  it('does not dispatch fetchUserAccount if username does not change', () => {
+    const username = 'black-panther';
+    const params = { username };
+
+    const { store } = signInUserWithUsername(username);
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+
+    const root = renderUserProfileEdit({ params, store });
+
+    dispatchSpy.reset();
+
+    root.setProps({ params });
+
+    sinon.assert.notCalled(dispatchSpy);
+  });
+
+  it('renders a username input field', () => {
+    const username = 'some username';
+    const root = renderUserProfileEdit({
+      params: { username },
+      userProps: {
+        ...defaultUserProps,
+        username,
+      },
+    });
+
+    expect(root.find('.UserProfileEdit-username')).toHaveLength(1);
```

Comment:
I think we've discussed this before, perhaps with you saying the opposite on one of my PRs, but is it really necessary to have the `toHaveLength(1)` test here when we're also doing a `toHaveProp('value', username)` check immediately below? It just seems redundant.

Reply:
To me, there are two intents: first making sure that we show a field, and then asserting one of its props.

Reply:
Right, but the second will definitely fail if the field isn't present, so we don't really need the first. But I'm fine leaving it in if you think it adds value.

Reply:
One more thing: if the second line fails, you have to figure out why and there are two options. With two different assertions, you know why it fails directly.